Avionics Anonymous UAVCAN Peripherals

avionicsanonymous opened the pull request:
Adds pages for the Avionics Anonymous line of UAVCAN peripherals, and adds them to the positioning, rangefinder, and UAVCAN peripheral lists.

Hwurzburg commented on common/source/docs/common-avanon-laserint.rst:
```diff
@@ -0,0 +1,26 @@
+
+   Avionics Anonymous Laser Interface
+
+The following parameter should be set on the autopilot (and then reboot the autopilot):
```
parameters

Hwurzburg commented on common/source/docs/common-avanon-gnss-mag.rst:
```diff
@@ -0,0 +1,25 @@
+
+   Avionics Anonymous GNSS/Mag: GPS and Compass
+
+The following parameter should be set on the autopilot (and then reboot the autopilot):
```
parameters

Hwurzburg commented on common/source/docs/common-avanon-mag.rst:
```diff
@@ -0,0 +1,23 @@
+
+=========================================
+Avionics Anonymous Magnetometer - Compass
+=========================================
```
Mgnetometer or Compass    choose one, the other is redundant...

Hwurzburg commented on common/source/docs/common-avanon-mag.rst:
```diff
@@ -0,0 +1,23 @@
+=========================================
+
+The `Avionics Anonymous Magnetometer <https://docs.avionicsanonymous.com/devices/minimag>`__ is a high-performance
+magnetometer for small unmanned vehicles. Includes a high-quality Memsic MMC5983MA magnetometer enabling 
```
It includes...

Hwurzburg commented on common/source/docs/common-avanon-mag.rst:
```diff
@@ -0,0 +1,23 @@
+
+The `Avionics Anonymous Magnetometer <https://docs.avionicsanonymous.com/devices/minimag>`__ is a high-performance
+magnetometer for small unmanned vehicles. Includes a high-quality Memsic MMC5983MA magnetometer enabling 
+sub-1-degree heading accuracy. All of this interfaces to your autopilot via robust `UAVCAN <https://uavcan.org>`__ 
```
All of these interface

Hwurzburg commented on common/source/docs/common-positioning-landing-page.rst:
```diff
@@ -14,6 +14,8 @@ positioning technologies:
 .. toctree::
     :maxdepth: 1
     
+    Avionics Anonymous GNSS/Mag: GNSS + Compass <common-avanon-gnss-mag>
```
Drop GNSS/Mag     GNSS + Compass is sufficient

Hwurzburg commented on common/source/docs/common-positioning-landing-page.rst:
```diff
@@ -14,6 +14,8 @@ positioning technologies:
     :maxdepth: 1
     
+    Avionics Anonymous GNSS/Mag: GNSS + Compass <common-avanon-gnss-mag>
+    Avionics Anonymous Magnetometer: Compass <common-avanon-mag>
```
Same here: Compass or Magnetometer, not both

Hwurzburg commented on common/source/docs/common-rangefinder-landingpage.rst:
```diff
@@ -53,6 +53,7 @@ based upon your set-up.
 
     Aerotenna US-D1 Radar Altimeter <common-aerotenna-usd1>
     Attollo Engineering Wasp200 <common-wasp200-lidar>
+    Avionics Anonymous Laser Interface <common-avanon-laserint>
```
Name change as above please

Hwurzburg commented on common/source/docs/common-uavcan-peripherals.rst:
```diff
@@ -4,7 +4,10 @@
-
+- `Avionics Anonymous Air Data Computer <https://docs.avionicsanonymous.com/devices/microadc>`__
+- `Avionics Anonymous GNSS/Mag <https://docs.avionicsanonymous.com/devices/gnssmag>`__
+- `Avionics Anonymous Laser Interface <https://docs.avionicsanonymous.com/devices/laser_interface>`__
```
name change please as requested above

Hwurzburg reviewed (changes requested):
the -adc page is not in any TOC, but dont worry about it...after you fix these, I will rebase, squash , and redo the UAVCAN peripherals page to a TOC format and add it there....and push to your PR, then merge....thanks

avionicsanonymous commented:
@Hwurzburg , thanks for the thorough review! I believe I've addressed all of your comments.

Sorry for for harassing I2C if it felt like too much "marketing" - a bit too much copy/paste from my other pages.

avionicsanonymous commented:
> 
> 
> the -adc page is not in any TOC, but dont worry about it...after you fix these, I will rebase, squash , and redo the UAVCAN peripherals page to a TOC format and add it there....and push to your PR, then merge....thanks

Thanks! I added the page but couldn't find any section at all for airspeed sensors, so figured I'd leave it orphaned until such a page was added.

Hwurzburg commented on common/source/docs/common-avanon-adc.rst:
```diff
@@ -0,0 +1,25 @@
+richly-featured Air Data Computer (ADC) for small unmanned airplanes. Includes an MS5611 barometric 
+pressure sensor for 10cm-resolution altitude, a Honeywell RSC-series differential pressure transducer 
+for extremely accurate airspeed, and a thermistor interface with high-resolution convertor for air 
+temperature measurement with less than 1 degree precision. All of this interfaces to your autopilot 
```
first change not doen

Hwurzburg commented on common/source/docs/common-avanon-gnss-mag.rst:
```diff
@@ -1,21 +1,21 @@
-interface - no sketchy I2C wiring for your magnetometer! 
+combination GPS and magnetometer for small unmanned vehicles. It includes a uBlox SAM-M8Q integrated 
+multi-constellation GNSS unit and a high-quality Memsic MMC5983MA magnetometer enabling better than 1-degree 
+heading accuracy. The devices interfaces to your autopilot via a robust `UAVCAN <https://uavcan.org>`__ 
```
These devices interface to ... 

or This device interfaces to...

Hwurzburg commented on common/source/docs/common-avanon-mag.rst:
```diff
@@ -1,18 +1,18 @@
-sub-1-degree heading accuracy. All of this interfaces to your autopilot via robust `UAVCAN <https://uavcan.org>`__ 
-interface - no sketchy I2C wiring for your magnetometer! 
+magnetometer for small unmanned vehicles. It includes a high-quality Memsic MMC5983MA magnetometer enabling 
+better than 1-degree heading accuracy. The devices interfaces to your autopilot via a robust `UAVCAN <https://uavcan.org>`__ 
```
This device interfaces to...

Hwurzburg commented:
I fixed the still outstanding change I requested, added a note on the rangefinder interface that I think would otherwise cause support calls to one of us, squash, rebased, changed the UAVCAN peripherals page to allow inclusion of the Airspeed sensor in a TOC, and merged....thanks
